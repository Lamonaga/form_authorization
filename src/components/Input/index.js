import './style.css'

const Input = ({onChange, value, id, name, type}) => {
    return (
        <>
            <input
                id={id}
                name={name}
                type={type}
                onChange={onChange}
                value={value}/>
        </>
    )
}

export default Input;