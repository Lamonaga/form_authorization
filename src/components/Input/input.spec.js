import React from "react";

import {shallow} from "enzyme";

import enzymeConfig from '../../enzymeConfig'
import Input from "./index";


const setUp = () => shallow(<Input/>);

	describe("Input component", () => {
		let component;
		let instance;

		beforeEach(() => {
			component = setUp();
			instance = component.instance();
		})

		it("should render Input conponent", () => {
			expect(component).toMatchSnapshot();
		})

		describe("input handle", () => {
			it("should handleChange search input", () => {
				const mockCallBack = jest.fn()
				const component = shallow(<Input onChange={mockCallBack}/> )
				expect(mockCallBack.mock.calls.length).toBe(0)
				component.find('input').simulate('change');
				expect(mockCallBack.mock.calls.length).toBe(1)
			})
		})
	})