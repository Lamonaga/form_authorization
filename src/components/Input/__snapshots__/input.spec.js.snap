// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render Input conponent 1`] = `ShallowWrapper {}`;
